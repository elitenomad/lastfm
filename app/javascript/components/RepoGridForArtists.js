import React from 'react';
import PropTypes from "prop-types";
import { Link } from 'react-router-dom';

function RepoGridForArtists ({ artists }) {
    return (
        <ul className='popular-list'>
            {artists.map(({name, image, url}, index) => {
                const tracks_url_per_artist = `/artists/${name.split(' ').join('+')}/tracks`
                return (
                    <li key={name} className='popular-item'>
                        <div className='popular-rank'>#{index + 1}</div>
                        <ul className='space-list-items'>
                            <li>
                                <img
                                    className='avatar'
                                    src={image[2]['#text']}
                                    alt={`Avatar for ${name}`}
                                />
                            </li>
                            <li>@{name}</li>
                            <li>{ url }</li>
                            <li>
                                <Link className='button' to={tracks_url_per_artist}>Top Tracks of {name}</Link>
                            </li>
                        </ul>
                    </li>
                )
            })}

            { artists.length <= 0 ? 'No results for the search string' : '' }
        </ul>
    )
}

RepoGridForArtists.propTypes = {
    artists: PropTypes.array.isRequired,
};


export default RepoGridForArtists;