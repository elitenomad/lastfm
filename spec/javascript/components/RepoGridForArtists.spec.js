import RepoGridForArtists from './../../../app/javascript/components/RepoGridForArtists';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('RepoGridForArtists Component', () => {
    const artists = [
        {
            "name":"David Bowie",
            "listeners":"3246761",
            "mbid":"5441c29d-3602-4898-b1a1-b77fa23b8e50",
            "url":"https://www.last.fm/music/David+Bowie",
            "streamable":"0",
            "image":[
                {
                    "#text":"https://lastfm-img2.akamaized.net/i/u/34s/aa9dfbdec965d751cd1fa4ec5a309299.png",
                    "size":"small"
                },
                {
                    "#text":"https://lastfm-img2.akamaized.net/i/u/64s/aa9dfbdec965d751cd1fa4ec5a309299.png",
                    "size":"medium"
                },
                {
                    "#text":"https://lastfm-img2.akamaized.net/i/u/174s/aa9dfbdec965d751cd1fa4ec5a309299.png",
                    "size":"large"
                },
                {
                    "#text":"https://lastfm-img2.akamaized.net/i/u/300x300/aa9dfbdec965d751cd1fa4ec5a309299.png",
                    "size":"extralarge"
                },
                {
                    "#text":"https://lastfm-img2.akamaized.net/i/u/300x300/aa9dfbdec965d751cd1fa4ec5a309299.png",
                    "size":"mega"
                }
            ]
        }
    ];

    it('is expected to render correctly', () => {
        const wrapper = shallow(<RepoGridForArtists  artists={artists}/>);
        expect(wrapper).toMatchSnapshot();

        const li = wrapper.find('.poplular-item');
        expect(li.find('ul.space-list-items')).not.toBeNull();
    });
});


