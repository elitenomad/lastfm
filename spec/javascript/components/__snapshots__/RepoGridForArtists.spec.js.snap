// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoGridForArtists Component is expected to render correctly 1`] = `
<ul
  className="popular-list"
>
  <li
    className="popular-item"
    key="David Bowie"
  >
    <div
      className="popular-rank"
    >
      #
      1
    </div>
    <ul
      className="space-list-items"
    >
      <li>
        <img
          alt="Avatar for David Bowie"
          className="avatar"
          src="https://lastfm-img2.akamaized.net/i/u/174s/aa9dfbdec965d751cd1fa4ec5a309299.png"
        />
      </li>
      <li>
        @
        David Bowie
      </li>
      <li>
        https://www.last.fm/music/David+Bowie
      </li>
      <li>
        <Link
          className="button"
          replace={false}
          to="/artists/David+Bowie/tracks"
        >
          Top Tracks of 
          David Bowie
        </Link>
      </li>
    </ul>
  </li>
</ul>
`;
